File: tests/testdata/relative_plugin_node/random.ts
// Copyright 2016-2024, KhulnaSoft Ltd.  All rights reserved.

import * as codeinfra from "@codeinfra/codeinfra";

export class Random extends codeinfra.Resource {
    result!: codeinfra.Output<string | undefined>;
  
    constructor(name: string, length: number, opts?: codeinfra.ResourceOptions) {
      const inputs: any = {};
      inputs["length"] = length;
      super("testprovider:index:Random", name, true, inputs, opts);
    }
  }

File: tests/integration/construct_component_methods_unknown/python/random_.py
# Copyright 2016-2021, KhulnaSoft Ltd.  All rights reserved.

from typing import Optional

import codeinfra

class Random(codeinfra.CustomResource):
    def __init__(self,
                 resource_name: str,
                 length: codeinfra.Input[int],
                 opts: Optional[codeinfra.ResourceOptions] = None):
        props = {
            "length": length,
            "result": None,
        }
        super().__init__("testprovider:index:Random", resource_name, props, opts)

    @property
    @codeinfra.getter
    def length(self) -> codeinfra.Output[int]:
        return codeinfra.get(self, "length")

    @property
    @codeinfra.getter
    def result(self) -> codeinfra.Output[str]:
        return codeinfra.get(self, "result")
